backend/app/services/nl_parser.py

```
@@ -102,39 +102,56 @@ def _get_lc_components():
         # Try OpenAI if key present, otherwise Ollama as default
-        if settings.OPENAI_API_KEY:
+        if api_key:
             try:
                 from langchain_openai import ChatOpenAI
                 llm = ChatOpenAI(
-                    model=settings.NL_DEFAULT_MODEL or "gpt-4o-mini",
+                    model=model or "gpt-4o-mini",
+                    api_key=api_key,
                     temperature=0.0,
                 )
             except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)